### PR TITLE
Require maven 3.9.8 to use airbase

### PR DESCRIPTION
**Pull request by wendigo**
3.9.8 is version used by mvnd 1.0.0 so let's make it a minimum required version which also means that we will support mvnd 1.0.0+ out of the box.

### CODE REVIEW

File: airbase/pom.xml [hunk 1]
```diff
@@ -148,7 +148,7 @@
         <air.check.fail-checkstyle>${air.check.fail-basic}</air.check.fail-checkstyle>
 
         <!-- minimum maven version to build -->
-        <air.maven.version>3.6.3</air.maven.version>
```
Comment (findepi) [inline airbase/pom.xml:151]: What's the benefit of not allowing to run the build with 3.6

File: airbase/pom.xml [hunk 2]
```diff
@@ -148,7 +148,7 @@
...
+        <air.maven.version>3.9.8</air.maven.version>
```
Comment (findepi) [inline airbase/pom.xml:151]: How old is 3.9.8?

Comment (wendigo) [reply airbase/pom.xml:151]: It's not about how old it is. I'd like to see mvnd supported across the board and 1.0.0 (reporting as 3.9.8) is the first version I actually trust to deliver a value

Comment (findepi) [reply airbase/pom.xml:151]: Age matters. if 3.9.8 is relatively recent and for example we uncover some edge case bug in 3.9.8 and there is no other version we can use, we actively prevented ourselves from using an older version as a workaround

Comment (wendigo) [reply airbase/pom.xml:151]: This can be overridden if needed with the property.

**Review by mosabua — APPROVED** [review]
Maven releases have been getting more and more stable and there are a lot of good refactors between 3.6.3 and 3.9.8. 

Definitely agree with this upgrade.

**Comment by wendigo** [comment]
That's my thinking. I've invested a lot of time into getting our maven infrastructure modern, simple, performant, and reproducible. Maven 3.9.8 along with the mvnd 1.0.0 (built on top of the 3.9.8) marks the culmination of this effort. I'd like to enforce the usage of these versions across our projects to have consistent behavior and also to claim that mvnd is supported and recommended for local development.

**Comment by wendigo** [comment]
@electrum @dain @martint please chime in

**Comment by wendigo** [comment]
The same discussion happens in the Quarkus community: https://groups.google.com/g/quarkus-dev/c/XU8i2tCEVPA

**Comment by findepi** [comment]
> The same discussion happens in the Quarkus community: https://groups.google.com/g/quarkus-dev/c/XU8i2tCEVPA

thanks for the link, this is really interesting!

I love how this thread starts with a clearly defined rationale:
_"We are not usually being too aggressive for our Maven requirements but... in this case we have a very good reason: Apache Maven 3.9.5- doesn't support Maven plugins compiled with Java 17."_

**Comment by wendigo** [comment]
@findepi I wasn't aware of that problem and I think that we only use plugins compiled with JDK8 (which is required for Maven 3.x)

**Review by electrum — APPROVED** [review]

**Comment by slachiewicz** [comment]
While updating Maven Wrapper to use the latest version of Maven is a good thing - requirement for all developers to use the latest version may be too aggressive.
Saying that You require Maven 3.9.x should be good enough.

In theory the Maven community supports Maven 3.6.3+ and we migrate all our base plugins to that version.

**Comment by mosabua** [comment]
@slachiewicz sure... but if we want to require 3.9.8 for airbase and any projects inheriting it would make our support for all those projects easier .. less testing and verification work. And also with wrapper config and such its completely transparent and easy to use whatever Maven version. Our move to require 3.9.8 is also explicitly to help the Maven project with testing and moving towards 4.0.0. We want to ensure that the latest and greatest Maven version is the best .. and dont really care about old stuff.

**Comment by electrum** [comment]
I support requiring the latest version. We can pretend that we support 3.6.3 but in reality we only test against the latest. Unless someone has a specific objection to upgrading (vs theoretical problems) then I’m good with this.

**Comment by mosabua** [comment]
Let's ship this @electrum and @wendigo !

**Pull request merged**